Sources/ClientRuntime/Networking/Streaming/DataStreamReader.swift

```diff
@@ -4,47 +4,62 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 //
+
 import AwsCommonRuntimeKit
 import Foundation
 
+/// A `StreamReader` that reads from a `ByteBuffer`
+/// It allows thread safe reading, writing and seeking
 public class DataStreamReader: StreamReader {
     private var _availableForRead: UInt
-
     public var availableForRead: UInt {
-        return _availableForRead
+        get {
```

Comment:
While we're here, can we convert `DataStreamReader` to an actor and remove the locks? It would simplify the implementation, make it less bug prone, and keeps the code updated with modern swift practices.

Reply:
almost forgot there are Actors now. 100%

Reply:
done

Reply:
Dropped the idea to use actor, because we have exposed StreamReader interface in public models which confirms to Codable.

Coddle is not async protocol and it is not possible to confirm it without a hacky solution.